### ISSUE TITLE
SIT - CMS_117_3M_6 - Mustang speed belt SPCMS , 3M H=6mm Lp=117

### ISSUE THREAD

**Issue by Lawproto** [icub-tech-iit/cad-libraries#88]
| Description          | Field                                  |
|:---------------------|:---------------------------------------|
| Datasheet component  | -                 |
| Project              | ergoCub lower arm    |
| Alias Creo           | CMS_117_3M_6                         |
| IIT-Code             | 15559                     |
| Applicant            | Lorenzo Protopapa, @Lawproto                |
| Supervisor           | Luca Fiorio, @fiorisi              |
| Maintainer           | Marcello Savoldi, @marcellosavoldi     |
| Coworkers            | @icub-tech-iit/silo-mech               |

### Notes:
**This is a flexible component, so I am not sure whether to store its 3D in cad-libraries representing the particular shape it assumes in the assembly I am working with or not.** And if not, which other shape to assume as default and how to handle the flexed shape in the assembly while still recalling the right component.

### Checklist:
Complete the checklist in order to proceed with the request.

- [ ] Created the Alias following [the standard convention](https://github.com/icub-tech-iit/cad-libraries/wiki/Mechanical-design-guidelines#commercial-components-coding-standard).
- [ ] Added, if necessary, the source files (STEP, IGS, etc), the links/datasheets/drawings or other information used to create the library part.
- [ ] File imported using the Creo [start part commercial](https://github.com/icub-tech-iit/cad-libraries/wiki/Mechanical-design-guidelines#standard-part-for-commercial-components).
- [ ] Checked [mass properties](https://github.com/icub-tech-iit/cad-libraries/wiki/Mechanical-design-guidelines#mass-properties-of-commercial-parts).
- [ ] Checked if one [special case](https://github.com/icub-tech-iit/cad-libraries/wiki/Mechanical-design-guidelines#special-cases-and-examples) applies to this component.
- [ ] Converted the files with the [EDU-COM license](https://github.com/icub-tech-iit/cad-libraries/wiki/PTC-Creo-Guidelines#save-file-with-commercial-license).
- [ ] Created IIT code in [WINGST](http://wingst.icub.iit.local/) with manufacturer and at least one supplier.

**Comment by marcellosavoldi** [comment]
How do you plan to proceed to make this belt?
In the past we have created a part with only the profile represented with a SKETCH.
then modeled the belt with the shape you needed and entered the wording "COSM" in the "TYPE" parameter
Then the profile and cosmetic model were inserted into an assembly, which was assembled on the pulleys.

**Comment by Lawproto** [comment]
@marcellosavoldi my intuition inspires me to go for two different parts, to be kept together as a Creo group inside the assembly for convenience.
- one is the `cad-libraries` part, with all the useful parameters filled in and a generic, non-flexed shape. This is what appears in the assembly BOM.
- one is a `COSM` part with the shape needed for the assembly to be consistent. No parameters, no appearance in BOM.

I am just not sure on whether to calculate the mass in the library part or the `COSM` part. In any case, the mass of the other one should result `0`.
How about it?

@icub-tech-iit/silo-mech

**Comment by marcellosavoldi** [comment]
the official part in CAD-LIBRARIES is modelled using a "cosmetic feature" (as surfaces or sketches) without setting a mass, but with the correct material.
For example a profile using a sketch or a 3D belt but created by surfaces features with mass=0 with type =TRAS
I would prefer the sketch because to hide a 3D object on the assembly I don't like...
... and in theory, by reading the wiki, it is forbidden to hide objects.
if it will be necessary, the exceptions on the WIKI must be indicated.

other opinions would be welcome... :-)

**Comment by Lawproto** [comment]
@marcellosavoldi I will keep your considerations into account. Thank you!

**Comment by marcellosavoldi** [comment]
keep in mind that most of the time we will also have a belt tensioner.
The position of the latter is not always the same.
I think could be better keeping the belt code to generate the BOM always in a bulk-item and design the belt as COSM.

**Comment by Lawproto** [comment]
@marcellosavoldi I am not sure I get your objection. As of now, the temporary belt CAD model does not take the tensioner into account, in fact there is interference.

![immagine](https://user-images.githubusercontent.com/84333301/160789909-e3bfa93d-1f29-442b-8ad7-8b633b3ba4d3.png)

I am fine to leave the representation like this, it is not a big deal for me. Taking belt tensioner into account seems too much work for too little gain.

**Comment by marcellosavoldi** [comment]
> @marcellosavoldi I am not sure I get your objection. As of now, the temporary belt CAD model does not take the tensioner into account, in fact there is interference.

Probably it's an error... a belt tensioner should always be present, unless the motor shaft can be moved before fastening it.

> I am fine to leave the representation like this, it is not a big deal for me. Taking belt tensioner into account seems too much work for too little gain.

probably if we had taken into account the belt tensioner we could have avoided the mistake.

**Comment by Lawproto** [comment]
@marcellosavoldi there is no mistake, the belt tensioner **is present**. It is just the belt that goes "straight" instead of following the very feeble deformation the belt tensioner would induce. To me, this simplification is perfectly fine.
Also, it is very difficult to foresee the entity of the belt deformation beforehand or to measure it afterhand, for that matter.

**Comment by marcellosavoldi** [comment]
>  It is just the belt that goes "straight" instead of following the very feeble deformation the belt tensioner would induce. 

I define this a error...

**Comment by Lawproto** [comment]
In the end I decided to give the library component a simple shape and to assemble it as flexible where required.
I put an annotation feature in the model tree explaining how to use it; the interaxis is dependent on the two diameters and the perimeter (not to be changed in value).

![immagine](https://user-images.githubusercontent.com/84333301/171869327-04271bd4-7e4b-4a01-aec3-b0851ba08f51.png)

Here is an example of application:

![immagine](https://user-images.githubusercontent.com/84333301/171869687-55078856-63f1-41ac-bc4b-3346d0658828.png)

Closing.
